modules/timepoint_list/php/timepoint_list.class.inc

```
@@ -300,5 +318,85 @@ class Timepoint_List extends \NDB_Menu
             )
         );
     }
-}
+    /**
+     * Handle archive a visitlabel
+     *
+     * @param ServerRequestInterface $request The PSR15 ServerRequestInterface.
+     *
+     * @return \LORIS\Http\Response\JSON\OK
+     */
+    protected function handleDelete(
+        ServerRequestInterface $request
+    ): ResponseInterface {
+        $body     = $request->getBody()->getContents();
+        $jsonBody = json_decode($body, true);
+        $id       = null;
+        if (is_array($jsonBody) && array_key_exists('sessionID', $jsonBody)) {
+            $id = $jsonBody['sessionID'];
+        }
+        $set = ['Active' => 'N'];
+        $where_sessionID = ['ID' => $id];
+        $DB = \NDB_Factory::singleton()->database();
+
+                $DB->update('session', $set, $where_sessionID);
+
+        return new \LORIS\Http\Response\JSON\OK(
+            ['success' =>'successfully archived']
+        );
+    }
+    /**
+     * Handle handlePost
+     *
+     * @param ServerRequestInterface $request The PSR15 handler.
+     *
+     * @return \LORIS\Http\Response\JSON\OK
+     */
+    protected function handlePost(ServerRequestInterface $request): ResponseInterface
+    {
+        $body     = $request->getBody()->getContents();
+        $jsonBody = json_decode($body, true);
+        // Check if $jsonBody is an array before accessing its keys
```

should return an error if not?